orbit/pkg/zenity/zenity.go

```diff
@@ -0,0 +1,126 @@
+//go:build linux
+
+package zenity
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os/exec"
+	"time"
+
+	"github.com/fleetdm/fleet/v4/server/contexts/ctxerr"
+)
+
+type Zenity struct {
+	// execCmdFn can be set in tests to mock execution of the dialog.
+	execCmdFn func(ctx context.Context, args ...string) ([]byte, int, error)
+}
+
+var (
+	ErrCanceled = errors.New("dialog canceled")
+	ErrTimeout  = errors.New("dialog timed out")
+	ErrUnknown  = errors.New("unknown error")
+)
+
+// EntryOptions represents options for the Entry dialog.
+type EntryOptions struct {
+	// Title sets the title of the dialog.
+	Title string
+
+	// Text sets the text of the dialog.
+	Text string
+
+	// HideText hides the text entered by the user.
+	HideText bool
+
+	// TimeOut sets the time in seconds before the dialog is automatically closed.
+	TimeOut time.Duration
+}
+
+// InfoOptions represents options for the Info dialog.
+type InfoOptions struct {
+	// Title sets the title of the dialog.
+	Title string
+
+	// Text sets the text of the dialog.
+	Text string
+
+	// Timeout sets the time in seconds before the dialog is automatically closed.
+	TimeOut time.Duration
+}
+
+// NewZenity creates a new Zenity dialog instance for zenity v4 on Linux.
+func NewZenity() *Zenity {
+	return &Zenity{
+		execCmdFn: execCmd,
+	}
+}
+
+// ShowEntry displays an dialog that accepts end user input. It returns the entered
+// text or errors ErrCanceled, ErrTimeout, or ErrUnknown.
+func (z *Zenity) ShowEntry(ctx context.Context, opts EntryOptions) ([]byte, error) {
+	args := []string{"--entry"}
+	if opts.Title != "" {
+		args = append(args, fmt.Sprintf(`--title="%s"`, opts.Title))
+	}
+	if opts.Text != "" {
+		args = append(args, fmt.Sprintf(`--text="%s"`, opts.Text))
```

Comment:
Should actually be:
```
args = append(args, "--title", opts.Title)
```
Otherwise the dialog has the double quotes for the given title and text. (Because of how Go escapes the arguments already.)

Reply:
good catch, i added `tools/dialog/main.go` to test visually on a host